### PR TITLE
Add product attribute generation to product page v2

### LINKED ISSUE #18949: [BO][IMPROVEMENT] Allow check all in product combinations
## Is your feature request related to a problem?

It is true that life expectancy has increased in recent years. But we still have to manage our time wisely.

#### Imagine the following situation : 

improve_combinaison

The user wishes to generate all the possible combinations (**sizes**, **colors**, **etc etc**)

- **size** : (4 values)
- **colors** : (14 values)
- **etc etc** : (N values)

In this case, the number of boxes he must check  =  **4 + 14 + N**

## Describe the solution you'd like

Consider the same situation but this time with a check all :

improve_combinaison (1)

In this case, the number of boxes he must check = **3**

### LINKED ISSUE #24261: [Experimental product page] Enable missing feature: generate combinations using the Combinations generator
### Describe the bug

The QA team explored the experimental product page and reported that generate combinations using the Combinations generator was missing.

#### Additional information

* PrestaShop version: `1.7.8.x`

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | This PR adds the combination generator inside a modal, with a tags search input with typehead and a scrollbar on attributes in order to avoid big modal's height
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/24261 and  https://github.com/PrestaShop/PrestaShop/issues/18949 and https://github.com/PrestaShop/PrestaShop/issues/12440 and https://github.com/PrestaShop/PrestaShop/issues/9939
| How to test?      | Go on product page v2, on combination, click on the button to generate combinations, play with the modal!
| Possible impacts? | Product combinations generation

# How does it looks?
Capture d’écran 2021-04-21 à 11 26 59

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23865)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Comment by PululuK** [comment]
Hi guys, very good job ! Thanks 
If you have the time, can you please add this feature please https://github.com/PrestaShop/PrestaShop/issues/18949
I can open a new PR if you want.

**Comment by NeOMakinG** [comment]
> Hi guys, very good job ! Thanks
> If you have the time, can you please add this feature please #18949
> I can open a new PR if you want.

Oh god, I love the idea, let's do it on another PR as this one is ready for review, as you want I may take it :)

**Comment by PululuK** [comment]
> > Hi guys, very good job ! Thanks
> > If you have the time, can you please add this feature please #18949
> > I can open a new PR if you want.
> 
> Oh god, I love the idea, let's do it on another PR as this one is ready for review, as you want I may take it :)

Ok bro ! Thanks

**Comment by NeOMakinG** [comment]
@PululuK as we discussed with @jolelievre we'll do it inside this PR

**Comment by PululuK** [comment]
> @PululuK as we discussed with @jolelievre we'll do it inside this PR

**Review by Progi1984 — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
@PululuK I added the checkbox feature :)

**Review by PierreRambaud — DISMISSED** [review]

**Review by PululuK — DISMISSED** [review]

File: admin-dev/themes/new-theme/js/pages/product/components/generator/AttributesSelector.vue
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/pages/product/components/generator/AttributesSelector.vue]:
```suggestion
        if (this.checkboxList.includes(attributeGroup)) {
          this.checkboxList = this.checkboxList.filter(
            (e) => e.id !== attributeGroup.id,
          );
        } else {
          this.checkboxList.push(attributeGroup);
        }
        this.$emit('toggleAll', {attributeGroup, select: !this.checkboxList.includes(attributeGroup)});
```

**Review by Progi1984 — COMMENTED** [review]
A last comment

**Review by matthieu-rolland — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by SD1982** [comment]
Hi @NeOMakinG 
The button to generate combination only appear after a first save off the product is this the desired behavior ?
I think it can be nice if the modal have an "autoclose" fonction when the combinations are generated what do you think ?
There are no pictures displayed in the combinations list
image
Also it's not really related with your PR but there is no bulk action for manipulate (delete, edit quantity)  the created combinations 
Except the points raised, the PR seems OK to me !!
Thanks @NeOMakinG  !!

**Comment by jolelievre** [comment]
Hi @SD1982,

I'm gonna try and answer your questions ^^

> The button to generate combination only appear after a first save off the product is this the desired behavior ?

Do you mean it's not available in the creation page? The yes it is expected, we can't create a combination until the product is created and has an ID. It used to be possible in legacy page because a "ghost" product was always created but it's a bad practice so now we wait for the first save and the ID to enable some features: combinations, images, ...

> I think it can be nice if the modal have an "autoclose" fonction when the combinations are generated what do you think?

This is doable, we didn't close it for now because I thought it may be confusing if the modal disappears and you don't remember which attributes were selected. By letting the merchant close the modal himself we reduce the risk for him to lose any info. But it's only my opinion your idea is also valid, so I'll let @TristanLDD and @MatShir decide about this.

> There are no pictures displayed in the combinations list

For existing combinations they should be here, for new combinations it's normal if they don't have some because we can't guess what images are adapted for each combination. We could choose one by default but I think there would much more chances to assign the bad image then the good one ^^ So by leaving no images associated, at least the merchant knows he needs to edit them to assign the images. An alternative could be to assign all product images? 🤷‍♂️. Here again I'll wait for the product team decision.

> Also it's not really related with your PR but there is no bulk action for manipulate (delete, edit quantity) the created combinations

Yes the bulk actions have not been implemented yet, they will be in next version only.

**Comment by MatShir** [comment]
Thanks for the feedback @SD1982, so:

* The image part is an issue related to the combination list. It is a regression from the list that can be fixed before the release or for the next version.

* We will test this way for the generation popup. We might change it depending on the feedbacks.

The rest was responded by @jolelievre.
If these were only the last blocking points, we can merge it then 😏

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @jolelievre & @SD1982

**Comment by PululuK** [comment]
Thanks guys !!!!

**Pull request merged**
